src/bin/common/solving_loop.ml

```diff
@@ -136,10 +136,10 @@ let main () =
           Options.Time.set_timeout (Options.get_timelimit ());
         end;
       SAT.reset_refs ();
-      let ftdn_env =
-        List.fold_left
-          (FE.process_decl ~hook_on_status:Frontend.print_status)
-          (FE.init_env used_context)
+      let ftdn_env = FE.init_env used_context in
+      let () =
+        List.iter
+          (FE.process_decl ~hook_on_status:Frontend.print_status ftdn_env)
           cnf
       in
```

Comment:
And I don't like `let () =` ever so where does that lead us :)

We don't have a style guide so I won't bikeshed, but you use the exact syntax I suggested in `worker_js.ml` so please at least be consistent.

Reply:
I agree that we should write style guide (not too long). I prefer using `let () =` too but it doesn't really matter.

Reply:
> We don't have a style guide so I won't bikeshed, but you use the exact syntax I suggested in worker_js.ml so please at least be consistent.

You do realize all I want now is to update `worker_js.ml`, do you? :rofl:

Reply:
More seriously, my first message was just explaining why I actually chose to use a `let ()` syntax, and I am sorry if you took it personally. I met people who liked `;` and others who liked `let ()`, and I can't guess what will be the opinion of my reviewer on the matter in advance.

If you want Alt-Ergo to follow a style guide for that matter, I will be happy to create it with you actually!

Reply:
> You do realize all I want now is to update worker_js.ml, do you? 🤣

I would like you to use the same format in both cases but as I said: we don't have a style guide currently, and so I won't bikeshed the alternative you choose — update either `worker_js.ml` or `solving_loop.ml` I won't care.

> If you want Alt-Ergo to follow a style guide for that matter, I will be happy to create it with you actually!

Having a style good sure would be a good thing (I guess #634 should be part of that discussion as well)! The existing code is a bit of a mess and mixes different styles in somewhat arbitrary ways. Opened #967

Reply:
Here is a proposition: as long as there is no rule about something, the decision goes to the person who wrote the code. We can't afford to lose hours of reflexion on stuff that are not specified.

Reply:
I am OK with that. I am not trying to force your hand in any way, and I have stated repeatedly that *I am OK with keeping `let () =` here*. I am simply asking that whatever style is used, the code at least be consistent within a single PR, is that really too much to ask?

Reply:
> I am not trying to force your hand in any way... 

You are.

Reply:
I think there is clearly a misunderstanding somewhere. I am sorry if this is the way your perceive what I am saying, because that is definitely not the intent.

Here is how the conversation went from my side:

 - I make a *suggestion* to change from `let () = ..` to using the sequence operator
 - You refuse to do it because you do not like the sequence operator for long expressions. That is fine!
 - I *accept that*, but point out that you did use the sequence operator in an exact copy of the same code, and ask you to be consistent (which is more important than any specific style being used)
 - You say "now I just want to update the other file!"
 - *I am OK with that*, which I state. I thought that we agreed, then, and that it was the end of it
 - The discussion switches to "what should we do when something is not specified" and to let the author make decisions
 - I agree with that (most of the time, style suggestions are that, *suggestions*), but there is a change that I thought we agreed on making (see above) and that was not done, which I point out, a bit sharply I admit, I am sorry for that.

To reiterate:
 - I have no strong feelings on the specific style being used
 - I do believe that it is important to be consistent to improve maintainability and clarity of the code
 - I want the PR to move forward based on a mutual understanding between author and reviewer
 - I am not going to hold up the PR for this (the only reason I have not approved yet is that I am waiting for the bench results)
 - If you prefer not to make any change here, please let me know (I guess that is the case, but again — this is not what I initially understood from the conversation); I am happy to move the PR forward based on mutual understanding.

I hope this helps clarify things, happy to have a chat with you one-on-one to clear things up if needed.

I understand that style preferences are subjective, and I respect your perspective on the matter. Looking forward to working with you on a style guide that can streamline these issues in the future :)

Reply:
Okay, I think I see where the misunderstanding is.

> You do realize all I want now is to update worker_js.ml, do you? 

This was a joke (which is what I meant in my next message starting with 'More seriously, ...') based on the fact that I believe the choice of `let ()` and `;` syntax can coexist without reducing readibility and I was making fun of my contradiction spirit :wink: I did not plan on updating `worker_js.ml` as I said, for that would be very aggressive way to express my stubbornness.

This conversation went a little too far :sweat_smile: sorry about that!

Reply:
Ah, I understand better now, jokes are not always easy to tell apart in text format! Glad we could clear that up.

> I did not plan on updating worker_js.ml as I said, for that would be very aggressive way to express my stubbornness.

I don't think it would have been aggressive, given that it was one of the two alternatives I was initially asking for.